pkg/client/request.go

```diff
@@ -302,45 +312,41 @@ func (v versionToResourceToFieldMapping) filterField(apiVersion, resourceType, f
 }
 
 var fieldMappings = versionToResourceToFieldMapping{
-	"v1beta1": resourceTypeToFieldMapping{
+	"v1beta3": resourceTypeToFieldMapping{
 		"nodes": clientFieldNameToAPIVersionFieldName{
-			ObjectNameField:   "name",
-			NodeUnschedulable: "unschedulable",
+			ObjectNameField:   "metadata.name",
+			NodeUnschedulable: "spec.unschedulable",
 		},
 		"minions": clientFieldNameToAPIVersionFieldName{
-			ObjectNameField:   "name",
-			NodeUnschedulable: "unschedulable",
+			ObjectNameField:   "metadata.name",
+			NodeUnschedulable: "spec.unschedulable",
 		},
 		"pods": clientFieldNameToAPIVersionFieldName{
-			PodHost: "DesiredState.Host",
+			PodHost: "spec.host",
 		},
 		"secrets": clientFieldNameToAPIVersionFieldName{
 			SecretType: "type",
 		},
 		"serviceAccounts": clientFieldNameToAPIVersionFieldName{
-			ObjectNameField: "name",
-		},
-	},
-	"v1beta2": resourceTypeToFieldMapping{
-		"nodes": clientFieldNameToAPIVersionFieldName{
-			ObjectNameField:   "name",
-			NodeUnschedulable: "unschedulable",
-		},
-		"minions": clientFieldNameToAPIVersionFieldName{
-			ObjectNameField:   "name",
-			NodeUnschedulable: "unschedulable",
+			ObjectNameField: "metadata.name",
 		},
-		"pods": clientFieldNameToAPIVersionFieldName{
-			PodHost: "DesiredState.Host",
+		"endpoints": clientFieldNameToAPIVersionFieldName{
+			ObjectNameField: "metadata.name",
 		},
-		"secrets": clientFieldNameToAPIVersionFieldName{
-			SecretType: "type",
-		},
-		"serviceAccounts": clientFieldNameToAPIVersionFieldName{
-			ObjectNameField: "name",
+		"events": clientFieldNameToAPIVersionFieldName{
+			ObjectNameField:              "metadata.name",
+			EventReason:                  "reason",
+			EventSource:                  "source",
+			EventInvolvedKind:            "involvedObject.kind",
+			EventInvolvedNamespace:       "involvedObject.namespace",
+			EventInvolvedName:            "involvedObject.name",
+			EventInvolvedUID:             "involvedObject.uid",
+			EventInvolvedAPIVersion:      "involvedObject.apiVersion",
+			EventInvolvedResourceVersion: "involvedObject.resourceVersion",
+			EventInvolvedFieldPath:       "involvedObject.fieldPath",
 		},
 	},
-	"v1beta3": resourceTypeToFieldMapping{
+	"v1": resourceTypeToFieldMapping{
```

Comment:
if I understand correctly, all of these things have to match:
* the label returned from the `AddFieldLabelConversionFunc` for Pod in all versions
* the key returned from `PodToSelectableFields`
* the value of `client.PodHost`